eng/common/scripts/Package-Properties.ps1

```diff
@@ -164,21 +159,45 @@ function Get-CSVMetadata ([string]$MetadataUri=$MetadataUri)
     return $metadataResponse
 }
 
-function Operate-OnPackages ($activePkgList, $ServiceDirectory, [Array]$pkgPropsResult)
+function Get-PkgPropsForEntireService ($serviceDirectoryPath, $serviceDirectoryName)
 {
-    foreach ($pkg in $activePkgList)
+    $projectProps = @() # Properties from very project inthe service
+    $packageProps = @() # Properties for artifacts specified in ci.yml
+
+    foreach ($directory in (Get-ChildItem $serviceDirectoryPath -Directory))
+    {
+        $pkgDirectoryPath = Join-Path $serviceDirectoryPath $directory.Name
+
+        if ($GetPackageInfoFromRepoFn -and (Test-Path "Function:$GetPackageInfoFromRepoFn"))
+        {
+            $pkgProps = &$GetPackageInfoFromRepoFn -pkgDirectoryPath $pkgDirectoryPath -serviceDirectory $serviceDirectoryName
+            if ($null -ne  $pkgProps)
+            {
+                $projectProps += $pkgProps
+            }
+        }
+        else
+        {
+            LogError "The function for '$GetPackageInfoFromRepoFn' was not found.`
+            Make sure it is present in eng/scripts/Language-Settings.ps1 and referenced in eng/common/scripts/common.ps1.`
+            See https://github.com/Azure/azure-sdk-tools/blob/master/doc/common/common_engsys.md#code-structure"
+        }
+    }
+
+    $ciYmlFiles = Get-ChildItem $serviceDirectoryPath -filter "ci.*yml"
+    foreach($ciYmlFile in $ciYmlFiles)
     {
-        LogDebug "Operating on $($pkg["name"])"
-        $pkgProps = Get-PkgProperties -PackageName $pkg["name"] -ServiceDirectory $ServiceDirectory
-        if ($null -ne  $pkgProps)
+        $activeArtifactList = Get-ArtifactListFromYml -ciYmlPath $ciYmlFile.FullName
+        foreach ($artifact in $activeArtifactList)
         {
-            $pkgPropsResult += $pkgProps
+            $packageProps += $projectProps | Where-Object { $_.ArtifactName -eq $artifact["name"] }
```

Comment:
You will need to be careful here with Java. In Java you also need to consider the group ID to disambiguate the artifact name.

Reply:
Is there any case where we have the same artifact name within the same service?